### PR TITLE
fix(header/p2p): return on context to avoid ugly errors on stop

### PR DESCRIPTION
**Pull request by Wondertan**
Self-explanatory

### CODE REVIEW

File: libs/header/p2p/session.go [hunk 1]
```diff
@@ -174,6 +176,7 @@ func (s *session[H]) doRequest(
 
 		select {
 		case <-s.ctx.Done():
+			return
```
Comment (vgonkivs) [inline libs/header/p2p/session.go:179]: Nice catch

**Review by vgonkivs — APPROVED** [review]

File: libs/header/p2p/session.go (hunk 1 above)
Comment (renaynay) [reply libs/header/p2p/session.go:179]: Wait didn't I mention this in one of the PRs?

**Review by renaynay — APPROVED** [review]

File: libs/header/p2p/session.go (hunk 1 above)
Comment (vgonkivs) [reply libs/header/p2p/session.go:179]: It seems that it was missed during resolving merge conflicts 😞

**Comment by vgonkivs** [comment]
I think we can stop using `area:header` label in `lib/header` related PRs in favour of `libs:header`.

**Pull request merged**